lib/ddtrace/diagnostics/environment_logger.rb

```diff
@@ -0,0 +1,268 @@
+        sampler = Datadog.configuration.tracer.sampler
+        return nil unless sampler
+
+        sampler.sample_rate(nil) rescue nil
+      end
+
+      # DEV: We currently only support SimpleRule instances.
+      # DEV: These are the most commonly used rules.
+      # DEV: We should expand support for other rules in the future,
+      # DEV: although it is tricky to serialize arbitrary rules.
+      #
+      # @return [Hash, nil] sample rules configured
+      def sampling_rules
+        sampler = Datadog.configuration.tracer.sampler
+        return nil unless sampler.is_a?(Datadog::PrioritySampler) &&
+                          sampler.priority_sampler.is_a?(Datadog::Sampling::RuleSampler)
+
+        sampler.priority_sampler.rules.map do |rule|
+          next unless rule.is_a?(Datadog::Sampling::SimpleRule)
+
+          {
+            name: rule.matcher.name,
+            service: rule.matcher.service,
+            sample_rate: rule.sampler.sample_rate(nil)
+          }
+        end.compact
+      end
+
+      # @return [Hash, nil] concatenated list of global tracer tags configured
+      def tags
+        tags = Datadog.configuration.tags
+        return nil if tags.empty?
+        hash_serializer(tags)
+      end
+
+      # @return [Boolean, nil] runtime metrics enabled in configuration
+      def runtime_metrics_enabled
+        Datadog.configuration.runtime_metrics.enabled
+      end
+
+      # Concatenated list of integrations activated, with their gem version.
+      # Example: "rails@6.0.3,rack@2.2.3"
+      #
+      # @return [String, nil]
+      def integrations_loaded
+        integrations = instrumented_integrations
+        return if integrations.empty?
+
+        integrations.map { |name, integration| "#{name}@#{integration.class.version}" }.join(','.freeze)
+      end
+
+      # Ruby VM name and version.
+      # Examples: "ruby-2.7.1", "jruby-9.2.11.1", "truffleruby-20.1.0"
+      # @return [String, nil]
+      def vm
+        # RUBY_ENGINE_VERSION returns the VM version, which
+        # will differ from RUBY_VERSION for non-mri VMs.
+        if defined?(RUBY_ENGINE_VERSION)
+          "#{RUBY_ENGINE}-#{RUBY_ENGINE_VERSION}"
+        else
+          # Ruby < 2.3 doesn't support RUBY_ENGINE_VERSION
+          "#{RUBY_ENGINE}-#{RUBY_VERSION}"
+        end
+      end
+
+      # @return [Boolean, nil] partial flushing enabled in configuration
+      def partial_flushing_enabled
+        !!Datadog.configuration.tracer.partial_flush.enabled
+      end
+
+      # @return [Boolean, nil] priority sampling enabled in configuration
+      def priority_sampling_enabled
+        !!Datadog.configuration.tracer.priority_sampling
+      end
+
+      # @return [Boolean, nil] health metrics enabled in configuration
+      def health_metrics_enabled
+        !!Datadog.configuration.diagnostics.health_metrics.enabled
+      end
+
+      # @return [Hash] environment information available at call time
+      def collect!(transport_responses)
```

Comment:
is either `logs_correlation_enabled` or `log_injection_enabled` something we're able to record? Perhaps just I see that listed in the google doc but is missing here.

Reply:
Also, just for posterity, we should make a `TODO` or put something in the backlog to add `profiling_enabled` here when that gets merged in

Reply:
We are not currently able to know if `logs_correlation_enabled`, as we don't support automatic log injection, and the snippets we suggest for users to implement it call generic functions. I'd added a placeholder for when we implement automatic log correlation injection.

I added a similar placeholder for profiling.